src/launchpad/kafka.py

```diff
@@ -41,51 +36,60 @@
 PREPROD_ARTIFACT_SCHEMA = get_codec(PREPROD_ARTIFACT_EVENTS_TOPIC)
 
 
-class LaunchpadMultiProcessingPool(MultiprocessingPool):
-    """Extended MultiprocessingPool with maxtasksperchild=1 to ensure clean worker state."""
+def _process_in_subprocess(decoded_message: Any, log_queue: multiprocessing.Queue[Any]) -> None:
+    """Worker function that runs in subprocess."""
+    root_logger = logging.getLogger()
+    root_logger.handlers.clear()
 
-    def maybe_create_pool(self) -> None:
-        if self._MultiprocessingPool__pool is None:
-            self._MultiprocessingPool__metrics.increment("arroyo.strategies.run_task_with_multiprocessing.pool.create")
-            self._MultiprocessingPool__pool = Pool(
-                self._MultiprocessingPool__num_processes,
-                initializer=partial(parallel_worker_initializer, self._MultiprocessingPool__initializer),
-                context=multiprocessing.get_context("spawn"),
-                maxtasksperchild=1,  # why we have this subclass
-            )
+    queue_handler = QueueHandler(log_queue)
+    queue_handler.addFilter(RequestLogFilter())
 
+    root_logger.addHandler(queue_handler)
+    root_logger.setLevel(logging.DEBUG)
 
-class LaunchpadRunTaskWithMultiprocessing(RunTaskWithMultiprocessing[TStrategyPayload, Any]):
-    """Tolerates child process exits from maxtasksperchild=1 by ignoring SIGCHLD."""
+    try:
+        ArtifactProcessor.process_message(decoded_message)
+    except Exception as e:
+        logger.exception(f"Error processing message in subprocess: {e}")
+        sys.exit(1)
 
-    def __init__(
-        self,
-        function: Callable[[Message[TStrategyPayload]], Any],
-        next_step: ProcessingStrategy[FilteredPayload | Any],
-        max_batch_size: int,
-        max_batch_time: float,
-        pool: MultiprocessingPool,
-        input_block_size: int | None = None,
-        output_block_size: int | None = None,
-    ) -> None:
-        super().__init__(function, next_step, max_batch_size, max_batch_time, pool, input_block_size, output_block_size)
-        # Override SIGCHLD handler - child exits are expected with maxtasksperchild=1
-        signal.signal(
-            signal.SIGCHLD,
-            lambda signum, frame: logger.debug(f"Worker process exited normally (SIGCHLD {signum})"),
-        )
 
+def process_kafka_message_with_service(
+    msg: Message[KafkaPayload],
+    log_queue: multiprocessing.Queue[Any],
+) -> Any:
+    """Process a Kafka message by spawning a fresh subprocess with timeout protection."""
+    timeout = int(os.getenv("KAFKA_TASK_TIMEOUT_SECONDS", "720"))  # 12 minutes default
```

Comment:
Maybe we should default to `None` for now to help us roll this out? That way the timeout is decoupled from the RunTaskInThreads change.